pallets/asset-registry/src/lib.rs

```diff
@@ -452,45 +504,45 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Convert Vec<u8> to BoundedVec so it respects the max set limit, otherwise return TooLong error
-	pub fn to_bounded_name(name: Vec<u8>) -> Result<BoundedVec<u8, T::StringLimit>, Error<T>> {
-		name.try_into().map_err(|_| Error::<T>::TooLong)
+	fn try_into_bounded(name: Option<Vec<u8>>) -> Result<Option<BoundedVec<u8, T::StringLimit>>, Error<T>> {
+		if let Some(name) = name {
+			TryInto::<BoundedVec<u8, T::StringLimit>>::try_into(name)
+				.map_err(|_| Error::<T>::TooLong)
+				.map(Some)
+		} else {
+			Ok(None)
+		}
 	}
```

Comment:
just a nitpick really, but i find this more readable.

```suggestion
	fn try_into_bounded(name: Option<Vec<u8>>) -> Result<Option<BoundedVec<u8, T::StringLimit>>, Error<T>> {
		let Some(s) = name else {
		    return Ok(None);
		}
		s.try_into().map_err(|_| Error::<T>::TooLong).map(Some)
		
	}
```